server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -299,7 +302,7 @@ export class AssetService {
         /**
          * Serve thumbnail image for both web and mobile app
          */
-        if (!query.isThumb && allowOriginalFile) {
+        if ((!query.isThumb && allowOriginalFile) || (query.isWeb && asset.mimeType === 'image/gif')) {
```

Comment:
I don't think you'll reach this statement, right? I think you need `asset.mimeType !== 'image/gif'` on line 282.